packages/generator-particle-base/src/generators/app/generatePromptOptions.ts

```diff
@@ -58,129 +48,88 @@ export const configurationPrompt = [
         return 'Please enter a two word project name with no spaces'
       }
       return true
-    },
+    }
   },
   {
     type: 'input',
     message:
-      'choose a component library name using kebab case. example "project-default"',
-    name: 'componentLibraryName',
-    default: 'project-default',
+      'Where does your design theme exist relative to the root of the project',
+    default: (answers: DesignTheme) =>
+      `./source/design/${answers.themeName}/`,
+    name: 'themePath'
+  }
+]
+
+const compileThemes = async (prev: DesignTheme[]) => {
+  return [...prev, await inquirer.prompt(designThemePrompt())]
+}
+
+const rerun = [{
+  type: 'confirm',
+  name: 'generateTheme',
+  message: 'Do you want to add another theme?',
+  default: true
+}]
+
+export const generatorLoop = async() => {
+  let loop = true;
+  let themesArray:DesignTheme[] = []
+  do {
+      themesArray = await compileThemes(themesArray)
+      loop = await inquirer.prompt(rerun).then(answers => answers.generateTheme)
```

Comment:
Why do we need a `Promise.resolve().then()` for await syntax? 

Also not sure what answers.generateTheme does. Can you explain it? Did a code search and it looks like it just sets the loop equal to a string.

Thinking something like this
```
answers = await inquirer.prompt(rerun)
loop = false
```

Reply:
It sets loop to a bool based on user response. Doing it this way because if I include that question in  `compileThemes ` I'd have to clean it out of every object in the themes array afterwards. 
The resolution of the promise is the then. w/o that I'd be setting the loop to a pending promise resolution.